File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/handleview/ActionButtonInterface.java
package net.kdt.pojavlaunch.customcontrols.handleview;

import android.view.View;

import net.kdt.pojavlaunch.customcontrols.buttons.ControlInterface;

/** Interface defining the behavior of action buttons */
public interface ActionButtonInterface extends View.OnClickListener {

    /** HAS TO BE CALLED BY THE CONSTRUCTOR */
    void init();

    /** Called when the button should be made aware of the current target */
    void setFollowedView(ControlInterface view);

    /** Called when the button action should be executed on the target */
    void onClick();

    /** Whether the button should be shown, given the current contextual information that it has */
    boolean shouldBeVisible();
/**
 * 「on Click」処理を実行します。
 * このメソッドは特定の機能を提供するために実装されています。
 */
    @Override  // Wrapper to remove the arg
    default void onClick(View v){onClick();}
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/handleview/ActionButtonInterface.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols.handleview

import android.view.View
import com.arata.yukarilauncher.ui.subassembly.customcontrols.buttons.ControlInterface

/** アクション行に配置されるアクションボタンのインターフェース。 */
interface ActionButtonInterface : View.OnClickListener {
    /** ボタンの初期化処理を行います。 */
    fun init()
    /**
     * フォローするビューを設定します。
     * @param view フォローするコントロール
     */
    fun setFollowedView(view: ControlInterface?)
    /** クリック時のアクションを実行します。 */
    fun onClick()
    /** @return ボタンを表示すべきかどうか */
    fun shouldBeVisible(): Boolean

    override fun onClick(v: View) = onClick()
}